ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStore.java

```diff
@@ -43,5 +45,19 @@
 
   public void putAll(MetaStore otherMetaStore);
 
+  public void addSourceNames(Set<String> sourceNames, String queryId);
+
+  public void addSinkNames(Set<String> sinkNames, String queryId);
+
+  public void removeQueryFromReferentialIntegrityTable(String queryId);
```

Comment:
perhaps rename to `removePersistentQuery` the name is leaking implementation detail